core/templates/base-components/oppia-footer.component.ts

```diff
@@ -72,9 +72,8 @@ export class OppiaFooterComponent {
   subscribeToMailingList(): void {
     this.mailingListBackendApiService.subscribeUserToMailingList(
       String(this.emailAddress),
-      // Most of the times, since the name is not provided
-      // so it is null. In such case placeholder name is set to UserX.
-      String(this.name) ?? 'UserX',
+      // Most of the time, the name is null since it is not provided.
+      String(this.name),
```

Comment:
I did a bit of digging and what I'm finding worries me ... it seems our "add_user_to_mailing_list" logic is a bit off and we have some users in the mailing list with name "null" because of that. Here is what should happen instead.

- If a name is not supplied, a null/None value for the name field should be passed to the backend instead.
- In the backend, this null value should be carried through, and not replaced with a string at any point.
- When we get to "add_user_to_mailing_list()" in user_services.py, the name field should be optional and default to None (so, move the "name" argument after the "tag" argument and fix all callsites). Any other calls to "add_user_to_mailing_list()" that supply an explicit name should have `name=` added before that argument since it now has a default value. Any calls that don't supply an explicit name should not pass that argument.
- In "add_user_to_mailing_list()", if the name is None then the merge_fields for name should not be provided to bulk_email_services.

Could you PTAL at this pipeline? We should fix this once and for all, otherwise this confusion about null/non-null names is going to continue.

Reply:
I will look into it and maybe update the PR by tomorrow evening.

Reply:
I have made some necessary changes, PTAL